### DISCUSSION TITLE
create-directus-project CLI: Allow "options" parameter for Postgresql and Cockroachdb connections

### DISCUSSION THREAD

**Discussion by jpizzle34** [directus/directus#15848] [Feature Requests]
When quickstarting a new project with:

```bash
pnpm create directus-project <my-project-name>
```
I would like to connect to a CockroachDB Serverless cluster which requires a specific cluster to be specified in the options parameter of the connection string:

e.g. `postgresql://<db-user>:<db-password>@free-tier8.aws-ap-southeast-1.cockroachlabs.cloud:26257/defaultdb?sslmode=verify-full&options=--cluster%3D<cluster-identifier>`

The problem is that the CLI only accepts db connection parameters and not a connection string, and the parameters that are asked for don't include an `options` string.

## Proposed Solution

1. Extend the Credentials interface to include an `options?: string` property:
https://github.com/directus/directus/blob/cbe1e7fbc69c8e9bd02254e9ed6efc35ba6468ef/api/src/cli/utils/create-db-connection.ts#L5-L14
2. Set the `options` property here exposing  `Knex.PgConnectionConfig`: https://github.com/directus/directus/blob/cbe1e7fbc69c8e9bd02254e9ed6efc35ba6468ef/api/src/cli/utils/create-db-connection.ts#L38-L41
3. Add an optional question in the cli asking for an `options` string if the dbclient is `pg` or `cockroachdb` here: https://github.com/directus/directus/blob/cbe1e7fbc69c8e9bd02254e9ed6efc35ba6468ef/api/src/cli/commands/init/questions.ts

## Why?
This will make it alot easier for developers new to Directus to bootstrap a new project, especially those wanting to connect to a multi-tennant, serverless database offering like CockroachDB Serverless.

## Alternative Solutions
1. Do a manual installation as outlined below, setting the `DB_CONNECTION_STRING` env var before running `pnpm directus bootstrap`: https://docs.directus.io/self-hosted/installation/manual.html
2. In the CLI starter, give the developer an option to specify a db connection string instead of parameters.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
